### PR TITLE
✨ Implement SampleWorker

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

Contains a sample implementation for an `ExternalWorker`. 
This should not be considered a canon-implementation, as this serves only demo- and testing- purposes.

But it can be a baseline on how a worker could access the `ExternalTaskAPI`.

PR: #1

## How can others test the changes?

The worker meant to run as a service that periodically checks for waiting `ExternalTasks`.
It requires the `ExternalTaskApiClient` as a dependency and should therefore be included in an application that comes with that dependency.

You can see the [ProcessEngineMeta integration tests](https://github.com/process-engine/process_engine_meta/blob/feature/implement_external_task/_integration_tests/src/fixture_providers/test_fixture_provider.ts#L67) for an example.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

**Comment by ElRaptorus** [comment]
https://github.com/process-engine/process_engine_runtime/issues/89

File: src/external_task_worker.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+export class ExternalTaskSampleWorker {
+
+  public config: any;
+
+  private _externalTaskApiClient: IExternalTaskApi;
+  private _httpClient: HttpClient;
+
+  private _intervalTimer: any;
+
+  private sampleIdentity: IIdentity = {
```
[the remark this answers is not in the document]
Comment (cmg-dev) [reply src/external_task_worker.ts]: @5ebastianMeier Where does this come from?

[the remark this answers is not in the document]
Comment (cmg-dev) [reply src/external_task_worker.ts]: Please add a `_` as prefix...

File: src/external_task_worker.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+  /**
+   * Processes the given ExternalTask.
+   *
+   * @async
+   */
+  private async _processExternalTask<TPayloadType, TResultType>(externalTask: ExternalTask<TPayloadType>): Promise<void> {
+
+    logger.info(`Processing ExternalTask ${externalTask.id}.`);
+    const externalTaskInvocation: Model.Activities.ExternalTaskInvocation = this._parseInvocation(externalTask.payload);
```
Comment (roblen45) [inline src/external_task_worker.ts]: Im externalTask.payload soll nach den requested changes in process_engine_contracts und process_engine_core nur noch der TPayload drin stehen

File: src/external_task_worker.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+      this.start();
+    }
+  }
+
+  /**
+   * Processes the given ExternalTask.
+   *
+   * @async
+   */
+  private async _processExternalTask<TPayloadType, TResultType>(externalTask: ExternalTask<TPayloadType>): Promise<void> {
```
Comment (roblen45) [inline src/external_task_worker.ts]: Generics in TPayload und TResult umbennen.

File: src/external_task_worker.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    logger.info(`Finished processing ExternalTask with ID ${externalTask.id}.`);
+
+  }
+
+  /**
+   * Returns some sample result with which to finish the ExternalTasks.
+   *
+   * @returns The sample result.
+   */
+  private _getSampleResult(payload: any): any {
```
Comment (roblen45) [inline src/external_task_worker.ts]: payload ist TPayload

**Pull request merged**